server/src/test/scala/controllers/UsersControllerSpec.scala

```diff
@@ -284,4 +285,203 @@ class UsersControllerSpec extends PlayPostgresSpec with LoginUtils {
+        email = email,
+        password = Password.trusted("test123..."),
+        captcha = Captcha.trusted("test")
+      )
+
+      createVerifyLoginUser(request, client).futureValue
+
+      val forgotPasswordRequest = ForgotPassword.Request(email, Captcha.trusted("test"))
+      when(captchaApi.verify(any[Captcha]())).thenReturn(Future.successful(false))
+
+      val error = client
+        .forgotPassword(forgotPasswordRequest)
+        .map(_ => "Success when failure expected")
+        .recover { case NonFatal(ex) =>
+          ex.getMessage
+        }
+        .futureValue
+
+      error must be("Invalid captcha, try again")
+
+      when(captchaApi.verify(any[Captcha]())).thenReturn(Future.successful(true))
+    }
+  }
+
+  "POST /reset-password" should {
+    "reset a password for a given user" in withApiClient { client =>
+      when(captchaApi.verify(any[Captcha]())).thenReturn(Future.successful(true))
+      val name = Name.trusted("wiringbits")
+      val email = Email.trusted("test1@email.com")
+      val request = CreateUser.Request(
+        name = name,
+        email = email,
+        password = Password.trusted("test123..."),
+        captcha = Captcha.trusted("test")
+      )
+
+      val response = createVerifyLoginUser(request, client).futureValue
+
+      val forgotPasswordRequest = ForgotPassword.Request(email, Captcha.trusted("test"))
+      client.forgotPassword(forgotPasswordRequest).futureValue
+
+      val resetPasswordRequest = ResetPassword.Request(UserToken(response.id), Password.trusted("test456..."))
+      client.resetPassword(resetPasswordRequest).futureValue
+
+      val loginRequest = Login.Request(
+        email = request.email,
+        password = Password.trusted("test456..."),
+        captcha = Captcha.trusted("test")
+      )
+      val loginResponse = client.login(loginRequest).futureValue
+
+      loginResponse.name must be(request.name)
+      loginResponse.email must be(request.email)
+    }
+
+    "return a token when a user tries to reset a password" in withApiClient { client =>
+      val name = Name.trusted("wiringbits")
+      val email = Email.trusted("test1@email.com")
+      val request = CreateUser.Request(
+        name = name,
+        email = email,
+        password = Password.trusted("test123..."),
+        captcha = Captcha.trusted("test")
+      )
+
+      val userId = createVerifyLoginUser(request, client).futureValue.id
+
+      val forgotPasswordRequest = ForgotPassword.Request(email, Captcha.trusted("test"))
+      client.forgotPassword(forgotPasswordRequest).futureValue
+
+      val resetPasswordRequest =
+        ResetPassword.Request(UserToken(userId), Password.trusted("test456..."))
+
+      val response = client
+        .resetPassword(resetPasswordRequest)
+        .futureValue
+
+      response mustNot be(ResetPassword.Response(None))
+    }
+
+    "return a None token when a nonexistent user tries to reset a password" in withApiClient { client =>
```

Comment:
This is not required, we must fail if this ever happens, if the token is valid the user must exist